### PR TITLE
refactor error handling rest api

### LINKED ISSUE #1019: No errors in batch response
Some batch responses do not contain errors. For example:

    $ curl 'http://localhost:6541/batch' -H 'Content-Type: application/json;charset=UTF-8' --data-binary '[{"method" : "POST","path" : "http://localhost:6541"}]'
    {"responses": [{"body": {}, "code": 403}], "updated_resources": {"created": [], "modified": [], "removed": [], "changed_descendants": []}}

I would have expected that there were one or more errors in `responses.body.errors`.

This might be connected to #256.

### LINKED ISSUE #1277: too much noise in console output when running bin/polytester
The main reason for this is that we always include pyramid_exclog.
to make sure  every backend exception is logged.

We should only include this in productive installations (just move the include to the ini file)

### PR DESCRIPTION
**Pull request by joka**
API Changens:

Batch request errors:
   - same error descriptions like without batch (before only 400 BadRequest)

Request errors:
   - all HTTPClientErrors have json body with error description (before only 400, 403, 410 and 500)

Internal Changes:

   - mv request error handling / logging to rest.exceptions (SRP)
   - add json body with description to all HTTPClientErrors Fix #1019
   - make sure all request headers, body, errors entries are logged nicely on error 
   - rm dependency to cornice (reduce complexity)
   - reduce noise in bin/polytester output Fix #1277
   - Fixup: passwords in batch requests may appear in log file  #1407 
   - Fix: arbitrary long request body may appear in log file, max length is now  5000 chars
   - add global test fixture request_ 
   - add global test fixture log to capture log messages

### CODE REVIEW

**Comment by joka** [comment]
Still to do:  Test coverage is broken

**Comment by joka** [comment]
juhu my first green mark

**Comment by xi** [comment]
I think the frontend needs adjustments. I will talk to @joka about this today.

@joka: Is this somehow related to #256?

File: src/adhocracy_core/adhocracy_core/rest/test_exceptions.py [hunk 1]
```diff
@@ -3,9 +3,140 @@
...
+        request_.body = '[]'
+        self.make_one([], request_)
+        assert '[]' in str(log)
+
+    def test_log_ignore_if_request_body_json_other(self, request_, log):
+        request_.body = b'None'
+        self.make_one([], request_)
+        assert 'None' not in str(log)
+
+    def test_log_abbrivated_request_body_if_gt_5000(self, request_, log):
```
Comment (pallix) [inline src/adhocracy_core/adhocracy_core/rest/test_exceptions.py]: abbrivated => abbreviated

**Comment by joka** [comment]
> @joka can you maybe comment here what you explained to me?

ok

**Comment by xi** [comment]
I adapted the frontend and potentailly fixed #748 along the way. However, I still found some issues in the backend:

-   The status text for batch responses is always "Bad Request" even if the status code is 404 or 410.
-   The `description` in 404 responses is the requested path. Something like "The requested resource {path} could not be found." would be much nicer IMHO.
-   The response body to an OPTIONS request to a deleted resource is `{"errors":[{"location":"body","name":null,"description":"Invalid JSON request body"}],"status":"error"}`
-   The response body to an OPTIONS request to a non existing resource is empty

I did not do any structured testing. So this is only what I found be accident.

**Comment by joka** [comment]
> The status text for batch responses is always "Bad Request" even if the status code is 404 or 410.
fixed
>     The description in 404 responses is the requested path. Something like "The requested resource {path} could not be found." would be much nicer IMHO.
Now its the status code / title, more description is no available.
>     The response body to an OPTIONS request to a deleted resource is {"errors":[{"location":"body","name":null,"description":"Invalid JSON request body"}],"status":"error"}
cannot reproduce, can you show me?
>     The response body to an OPTIONS request to a non existing resource is empty
cannot reproduce, can you show me?

**Comment by xi** [comment]
>> The response body to an OPTIONS request to a deleted resource is {"errors":[{"location":"body","name":null,"description":"Invalid JSON request body"}],"status":"error"}
>
> cannot reproduce, can you show me?

I checked again and noticed that I only get this error when I add this header to the request: `Content-Type: application/json`.

>> The response body to an OPTIONS request to a non existing resource is empty
>
> cannot reproduce, can you show me?

Cannot reproduce anymore, too.

**Comment by joka** [comment]
> The response body to an OPTIONS request to a deleted resource is {"errors":[{"location":"body","name":null,"description":"Invalid JSON request body"}],"status":"error"}
> cannot reproduce, can you show me?
> I checked again and noticed that I only get this error when I add this header to the request: Content-Type: application/json.

This is the intended behaviour.  If you set the content type to json, the backend expects a valid json body

**Comment by xi** [comment]
>>> The response body to an OPTIONS request to a non existing resource is empty
>>
>> cannot reproduce, can you show me?
>
> Cannot reproduce anymore, too.

I had this issue again. However, it only showed up in chromium. Not in firefox or curl. So I guess it is a chromium bug.

**Pull request merged**